drivers/block-nbd.c

```diff
@@ -683,33 +650,23 @@ tdnbd_nbd_negotiate_old(struct tdnbd_data *prv, td_driver_t *driver)
 	driver->info.sector_size = DEFAULT_SECTOR_SIZE;
 	driver->info.info = 0;
 
-	if (tdnbd_wait_read(sock) <= 0) {
-		ERROR("Timeout in nbd_negotiate");
-		close(sock);
-		return -1;
+	rc = tdnbd_wait_recv(sock, &flags, sizeof(flags), 0);
+	if (rc < 0) {
+		ERROR("Error in nbd_negotiate: %s", strerror(-rc));
+		goto errout;
 	}
-
-	rc = recv(sock, &flags, sizeof(flags), 0);
 	if (rc < sizeof(flags)) {
-		ERROR("Short read in negotiation(4) (%d)\n", rc);
-		close(sock);
-		return -1;
+		ERROR("Short read in OLD negotiation(4) (%d)\n", rc);
+		goto errout;
 	}
 
 	INFO("Got flags: %"PRIu32"", ntohl(flags));
 
 	while (padbytes > 0) {
-		if (tdnbd_wait_read(sock) <= 0) {
-			ERROR("Timeout in nbd_negotiate");
-			close(sock);
-			return -1;
-		}
-
-		rc = recv(sock, buffer, padbytes, 0);
+		rc = tdnbd_wait_recv(sock, buffer, padbytes, 0);
 		if (rc < 0) {
-			ERROR("Bad read in negotiation(5) (%d)\n", rc);
-			close(sock);
-			return -1;
+			ERROR("Error in nbd_negotiate: %s", strerror(-rc));
+			// goto errout;
```

Comment:
Should the commented-out code be restored?

Reply:
Yes. Looks like a "comment this line" hotkey combination got activated without me noticing...